clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -222,4 +268,30 @@ private static String defaultKerberosRealm() throws ClassNotFoundException, NoSu
         getDefaultRealmMethod = classRef.getDeclaredMethod("getDefaultRealm", new Class[0]);
         return (String) getDefaultRealmMethod.invoke(kerbConf, new Object[0]);
     }
+
+    private void createClientCallbackHandler(Map<String, ?> configs, String mechanism) {
+        Class<? extends AuthenticateCallbackHandler> clazz = (Class<? extends AuthenticateCallbackHandler>) configs.get(SaslConfigs.SASL_CLIENT_CALLBACK_HANDLER_CLASS);
```

Comment:
See comment below about the order of built-in mechanisms vs. non-built-in ones.

Reply:
Same as before, callback handlers can be overridden for built-in mechanisms too.